src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp

```diff
@@ -150,11 +150,13 @@ HandleTransposeBeforeMatMul::HandleTransposeBeforeMatMul() {
             return false;
         }
 
-        auto matmul_node = matmul_iter->second.get_node_shared_ptr();
+        auto matmul_node = std::dynamic_pointer_cast<ngraph::opset8::MatMul>(matmul_iter->second.get_node_shared_ptr());
+        if (!matmul_node)
+            return false;
```

Comment:
Why is this required here? We shouldn't be here with a layer other than MatMul, because of pattern_map search above, right?